gobblin-runtime/src/main/java/org/apache/gobblin/util/MySQLStoreUtils.java

```diff
@@ -0,0 +1,70 @@
+package org.apache.gobblin.util;
+
+import com.zaxxer.hikari.HikariDataSource;
+import java.io.IOException;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import javax.sql.DataSource;
+import org.slf4j.Logger;
+
+
+/**
+ * MySQL based implementations of stores require common functionality that can be stored in a utility class. The
+ * functionality includes executing prepared statements on a data source object and executing SQL queries at fixed
+ * intervals. 
+ */
+public class MySQLStoreUtils {
+  private final DataSource dataSource;
```

Comment:
javadoc ought to delineate whose responsibility for resource mgmt of `DataSource`--this class or caller/user

Reply:
Addressed in comments

Reply:
would be clearer to say something like "MUST maintain ownership of the {@link DataSource} and arrange for it to be closed, but only once this instance will no longer be used"

... and with that in mind, how would the repeating SQL mesh with that?  should this instance hold on to what it schedules, and provide a `.close()` method that would unschedule them all?  if so, the advised shutdown protocol might be:
```
storeUtils.close();
dataSource.close();
```
?

Reply:
That's a good pt. I added a close method to unschedule any existing executor threads and close them.